### PR TITLE
Update cere node version

### PR DESCRIPTION
**Pull request by DmitriyMolch**
https://cerenetwork.atlassian.net/browse/CBI-2289

### CODE REVIEW

File: chains/substrate/test_helper_test.go [hunk 1]
```diff
@@ -11,11 +11,11 @@ import (
 	"github.com/Cerebellum-Network/chainbridge-utils/keystore"
 	"github.com/Cerebellum-Network/chainbridge-utils/msg"
 	"github.com/ChainSafe/log15"
-	"github.com/centrifuge/go-substrate-rpc-client/v2/types"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
-const TestEndpoint = "ws://127.0.0.1:9944"
+const TestEndpoint = "wss://ext-devs-node-3.cluster-3.cere.network:9945"
```
Comment (AndreiNavoichyk) [inline chains/substrate/test_helper_test.go]: Revert back before merge

File: chains/substrate/test_helper_test.go [hunk 2]
```diff
@@ -30,9 +30,13 @@ var BobTestLogger = newTestLogger("Bob")
 var ThisChain msg.ChainId = 1
 var ForeignChain msg.ChainId = 2
 
+// todo: understand how to manage it
+var aliceAccountId, _ = types.NewAccountID(AliceKey.PublicKey)
```
Comment (AndreiNavoichyk) [inline chains/substrate/test_helper_test.go:33]: What do you mean by `manage` here?

File: chains/substrate/types.go [hunk 3]
```diff
@@ -61,7 +62,10 @@ func (p *proposal) encode() ([]byte, error) {
 func (w *writer) createFungibleProposal(m msg.Message) (*proposal, error) {
 	bigAmt := big.NewInt(0).SetBytes(m.Payload[0].([]byte))
 	amount := types.NewU128(*bigAmt)
-	recipient := types.NewAccountID(m.Payload[1].([]byte))
+	recipient, err := types.NewAccountID(m.Payload[1].([]byte))
+	if err != nil {
```
Comment (AndreiNavoichyk) [inline chains/substrate/types.go:66]: Is it an API change?

File: chains/substrate/types.go [hunk 4]
```diff
@@ -97,7 +101,10 @@ func (w *writer) createFungibleProposal(m msg.Message) (*proposal, error) {
 
 func (w *writer) createNonFungibleProposal(m msg.Message) (*proposal, error) {
 	tokenId := types.NewU256(*big.NewInt(0).SetBytes(m.Payload[0].([]byte)))
-	recipient := types.NewAccountID(m.Payload[1].([]byte))
+	recipient, err := types.NewAccountID(m.Payload[1].([]byte))
+	if err != nil {
```
Comment (AndreiNavoichyk) [inline chains/substrate/types.go:105]: Is it an API change?

File: chains/substrate/types.go (hunk 3 above)
Comment (DmitriyMolch) [reply chains/substrate/types.go:66]: yes, it's a constructor interface change

File: chains/substrate/test_helper_test.go (hunk 2 above)
Comment (DmitriyMolch) [reply chains/substrate/test_helper_test.go:33]: Manage means  how to fix broken tests

File: chains/substrate/test_helper_test.go (hunk 1 above)
Comment (DmitriyMolch) [reply chains/substrate/test_helper_test.go]: ok

File: chains/substrate/types.go (hunk 4 above)
Comment (DmitriyMolch) [reply chains/substrate/types.go:105]: yes, it's a constructor interface change

File: Makefile [hunk 5]
```diff
@@ -77,7 +77,7 @@ install-subkey:
 ## Runs go test for all packages except the solidity bindings
 test:
 	@echo "  >  \033[32mRunning tests...\033[0m "
-	go test -p 1 -coverprofile=cover.out -v `go list ./... | grep -v bindings | grep -v e2e`
+	go test -p 1 -coverprofile=cover.out -v `go list ./... | grep -v bindings | grep -v e2e` --timeout 0
```
Comment (AndreiNavoichyk) [inline Makefile:80]: Why do we need it?

File: chains/substrate/connection.go [hunk 6]
```diff
@@ -181,16 +182,11 @@ func (c *Connection) queryStorage(prefix, method string, arg1, arg2 []byte, resu
...
-	for _, mod := range meta.AsMetadataV12.Modules {
-		if string(mod.Name) == prefix {
-			for _, cons := range mod.Constants {
-				if string(cons.Name) == name {
-					return types.DecodeFromBytes(cons.Value, res)
-				}
-			}
-		}
+	consValue, err := meta.AsMetadataV14.FindConstantValue(types.NewText(prefix), types.NewText(name))
+	if(err != nil) {
```
Comment (AndreiNavoichyk) [inline chains/substrate/connection.go]: Fix formatting

File: chains/substrate/test_helper_test.go [hunk 7]
```diff
@@ -93,8 +96,8 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		panic(err)
 	}
-	alice := NewWriter(aliceConn, AliceTestLogger, wSysErr, nil, true)
-	bob := NewWriter(bobConn, BobTestLogger, wSysErr, nil, true)
+	alice := NewWriter(aliceConn, AliceTestLogger, wSysErr, nil, false)
```
Comment (AndreiNavoichyk) [inline chains/substrate/test_helper_test.go:99]: Why did you change `true` to `false`?

File: config/config.json.example [hunk 8]
```diff
@@ -5,7 +5,7 @@
       "type": "ethereum",
       "id": "0",
       "endpoint": "https://rpc.goerli.mudit.blog",
-      "from": "0x9A9Dc7B059E9Add0D2d154A17139d8f2c4B38f6f",
+      "from": "0x19631d940Cc52CB5533eAbFa66bE139099410efE",
```
Comment (AndreiNavoichyk) [inline config/config.json.example:8]: Revert back

File: go.mod [hunk 9]
```diff
@@ -7,13 +7,18 @@ require (
...
+	github.com/prometheus/client_golang v1.13.0
+	github.com/prometheus/client_model v0.2.0 // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/urfave/cli/v2 v2.3.0
+	golang.org/x/crypto v0.1.0 // indirect
 )
 
-replace github.com/centrifuge/go-substrate-rpc-client/v2 v2.0.1 => github.com/Cerebellum-Network/go-substrate-rpc-client/v2 v2.0.2-cere
\ No newline at end of file
+replace github.com/Cerebellum-Network/chainbridge-substrate-events => ../chainbridge-substrate-events
```
Comment (AndreiNavoichyk) [inline go.mod]: Remove

File: e2e/fungible_test.go [hunk 10]
```diff
@@ -106,6 +106,7 @@ func testErc20ToErc20(t *testing.T, ctx *testContext) {
 }
 
 func testErc20SubstrateRoundTrip(t *testing.T, ctx *testContext) {
+	t.Skip("Skipped it until AccountInfo type will be fixed in the go-substrate-rpc-client repo")
```
Comment (AndreiNavoichyk) [inline e2e/fungible_test.go]: Add link to the issue on GitHub

File: Makefile (hunk 5 above)
Comment (DmitriyMolch) [reply Makefile:80]: To omit situation when tests fail by timeout

File: chains/substrate/test_helper_test.go (hunk 7 above)
Comment (DmitriyMolch) [reply chains/substrate/test_helper_test.go:99]: Because we need to send `useExtendedCall`=`false` to avoid the `Invalid Transaction` error

File: config/config.json.example (hunk 8 above)
Comment (DmitriyMolch) [reply config/config.json.example:8]: No need to do it as this configuration repeats the configuration from DEV. It will be required to run 2 relayers simultaneously.

File: e2e/fungible_test.go (hunk 10 above)
Comment (DmitriyMolch) [reply e2e/fungible_test.go]: Added

File: chains/substrate/connection.go (hunk 6 above)
Comment (DmitriyMolch) [reply chains/substrate/connection.go]: This is Github breaks the formating. It looks fine now in the code. But I need to fix linter because now it produces errors.

File: Makefile [hunk 11]
```diff
@@ -23,20 +23,20 @@ get:
...
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.46.2
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.50.1
 
 .PHONY: lint
 lint:
 	if [ ! -f ./bin/golangci-lint ]; then \
 		$(MAKE) get-lint; \
 	fi;
-	./bin/golangci-lint run ./... --timeout 5m0s
+	./bin/golangci-lint run  ./... --timeout 5m0s -v
```
Comment (AndreiNavoichyk) [inline Makefile]: Remove extra whitespaces

Comment (AndreiNavoichyk) [inline Makefile]: What is `-v`?

Comment (DmitriyMolch) [reply Makefile]: Fixed

File: Makefile [hunk 12]
```diff
@@ -23,20 +23,20 @@ get:
...
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.46.2
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.50.1
 
 .PHONY: lint
 lint:
 	if [ ! -f ./bin/golangci-lint ]; then \
 		$(MAKE) get-lint; \
 	fi;
-	./bin/golangci-lint run ./... --timeout 5m0s
+	./bin/golangci-lint run ./... --timeout 5m0s -v
```
Comment (AndreiNavoichyk) [inline Makefile:33]: What is `-v`?

Comment (DmitriyMolch) [reply Makefile:33]: `-v` means `verbose`. I added it to provide linting debug info.

File: go.mod [hunk 13]
```diff
@@ -14,6 +14,8 @@ require (
 	golang.org/x/crypto v0.1.0
 )
 
+replace github.com/centrifuge/go-substrate-rpc-client/v4 => github.com/Cerebellum-Network/go-substrate-rpc-client/v4 v4.0.9-cere
```
Comment (AndreiNavoichyk) [inline go.mod]: Why do you need this replace?

Comment (DmitriyMolch) [reply go.mod]: It gives less manual work: 
  1. You don't need to replace it in submodules.  
  2. There won't be any conflicts in the code in a case we want to merge something from upstream.

Comment (AndreiNavoichyk) [reply go.mod]: But it gives more hidden redirects. Let's get rid of it.

**Review by AndreiNavoichyk — COMMENTED** [review]
Update README

**Review by AndreiNavoichyk — APPROVED** [review]
